### ISSUE TITLE
How to upload the files generated by `python -m build` to pypi?

### ISSUE THREAD

**Issue by matsui528** [matsui528/rii#62]
When uploading the files generated by `python -m build` to pypi, I received [the following error](https://github.com/matsui528/rii/actions/runs/6845387876/job/18610516924). How to fix it?

Summary: 
```
Checking dist/rii-0.2.11-cp312-cp312-linux_x86_64.whl: PASSED
Checking dist/rii-0.2.11.tar.gz: PASSED
Uploading distributions to https://upload.pypi.org/legacy/
Uploading rii-0.2.11-cp312-cp312-linux_x86_64.whl
25l
  0% ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0.0/1.9 MB • --:-- • ?
  0% ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0.0/1.9 MB • --:-- • ?
  4% ━╸━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0.1/1.9 MB • 00:01 • 166.3 MB/s
  8% ━━━╺━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0.1/1.9 MB • 00:01 • 1.8 MB/s
 22% ━━━━━━━━╸━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0.4/1.9 MB • 00:01 • 1.9 MB/s
 22% ━━━━━━━━╸━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0.4/1.9 MB • 00:01 • 1.9 MB/s
 34% ━━━━━━━━━━━━━╸━━━━━━━━━━━━━━━━━━━━━━━━━━ 0.7/1.9 MB • 00:01 • 1.8 MB/s
100% ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.9/1.9 MB • 00:00 • 3.8 MB/s
100% ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.9/1.9 MB • 00:00 • 3.8 MB/s
100% ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.9/1.9 MB • 00:00 • 3.8 MB/s
100% ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.9/1.9 MB • 00:00 • 3.8 MB/s
100% ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.9/1.9 MB • 00:00 • 3.8 MB/s
100% ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.9/1.9 MB • 00:00 • 3.8 MB/s
100% ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.9/1.9 MB • 00:00 • 3.8 MB/s
100% ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.9/1.9 MB • 00:00 • 3.8 MB/s
25hWARNING  Error during upload. Retry with the --verbose option for more details. 
ERROR    HTTPError: 400 Bad Request from https://upload.pypi.org/legacy/        
         Binary wheel 'rii-0.2.11-cp312-cp312-linux_x86_64.whl' has an          
         unsupported platform tag 'linux_x86_64'.  
```

- Recently, I updated the build system to the new `pyproject.toml` #54 by following [the official pybind guideline](https://github.com/pybind/python_example/tree/master).
- Then, creating a distribution directory `dist` by `python setup.py sdist` is now deprecated. I replaced it with `python -m build`. #60 
- After that, it seems `python -m build` generates a file something like `rii-0.2.11-cp312-cp312-linux_x86_64.whl`. But pypi doesn't accept this name. It should be a simpler filename such as `rii-0.2.10.tar.gz` ? Ref: [previous successful file](https://pypi.org/project/rii/#files)

**Comment by kyamagu** [comment]
You'll need a platform-specific tag like `manylinux*`. Use [auditwheel](https://github.com/pypa/auditwheel), or build on [cibuildwheel](https://github.com/pypa/cibuildwheel) to resolve all the headaches around binary packages.

**Comment by matsui528** [comment]
Thank you! Let me take a look...

**Comment by matsui528** [comment]
Just adding `--sdist` option solved the problem 😂 #63
